### PR TITLE
Minor doc update to the consensus algorithms.

### LINKED ISSUE #13148: Better document consensus algorithms
#13082 adds a substantial amount of documentation to the NPX consensus algorithm implementation. The other consensus algorithm implementations could probably stand to gain from doing the same.

### PR DESCRIPTION
**Pull request by bangerth**
Fixes #13148.

/rebuild

### CODE REVIEW

File: include/deal.II/base/mpi_consensus_algorithms.h [hunk 1]
```diff
@@ -396,7 +395,10 @@ namespace Utilities
        * This class implements a concrete algorithm for the
        * ConsensusAlgorithms::Interface base class, using a two step approach.
        * In the first step the source ranks are determined and in the second
-       * step a static sparse data exchange is performed.
+       * step a static sparse data exchange is performed. This approach is
+       * efficient, but requires the computation of the communication
+       * pattern (see below); this algorithm is most suitable for relatively
```
Comment (peterrum) [inline include/deal.II/base/mpi_consensus_algorithms.h:400]: The problem is not that the communication pattern has to be computed, but how it is computed: global operation vs. p2p. 

_Side comment: Also I am not completely sure, if the name of the class is optimal. The "PEX" part used to be `compute_point_to_point_communication_pattern()`. Now `compute_point_to_point_communication_pattern()` also relies on "NBX" (or have introduced there the magic number as well to switch between implementations?). The class here organizes the data exchange in two steps._

Comment (bangerth) [reply include/deal.II/base/mpi_consensus_algorithms.h:400]: Hm, how do we address this? I *think* that your original idea was that this class is suitable for small problems, right? But as you note, the bottleneck for large problems was the `compute_p2p_communication_pattern()` function, which no longer has the bottleneck. The difference between the two implementations is simply whether they use two phases (NBX) or four phases (PEX: a two-phase NBX exchange plus the actual data exchange as send request + send answers).

I have to admit that I'm not sure I know the difference in intended use of the two algorithms well enough to describe this in the documentation :-)

Comment (peterrum) [reply include/deal.II/base/mpi_consensus_algorithms.h:400]: > I have to admit that I'm not sure I know the difference in intended use of the two algorithms well enough to describe this in the documentation :-)

I'll give it a thought. I'll come back at the earliest next week on this topic. Maybe if you drop this one comment, we could merge.

**Comment by bangerth** [comment]
I dropped the questionable sentence.

**Review by peterrum — APPROVED** [review]

**Comment by drwells** [comment]
It looks like `ready` is stuck - under "Details" everything appears to be in order so I'll merge.

**Pull request merged**